pkg/componentreadiness/jiraautomator/jiraautomator.go

```diff
@@ -0,0 +1,597 @@
+		values.Add("capability", reportOpts.TestIDOption.Capability)
+		values.Add("component", reportOpts.TestIDOption.Component)
+	} else if reportOpts.TestIDOption.Component != "" {
+		absURL += "env_capabilities?"
+		values.Add("component", reportOpts.TestIDOption.Component)
+	} else {
+		absURL += "main?"
+	}
+
+	if reportOpts.BaseRelease.Release != "" {
+		values.Add("baseRelease", reportOpts.BaseRelease.Release)
+		values.Add("baseStartTime", reportOpts.BaseRelease.Start.UTC().Format(time.RFC3339))
+		values.Add("baseEndTime", reportOpts.BaseRelease.End.UTC().Format(time.RFC3339))
+	}
+	if reportOpts.SampleRelease.Release != "" {
+		values.Add("sampleRelease", reportOpts.SampleRelease.Release)
+		values.Add("sampleStartTime", reportOpts.SampleRelease.Start.UTC().Format(time.RFC3339))
+		values.Add("sampleEndTime", reportOpts.SampleRelease.End.UTC().Format(time.RFC3339))
+	}
+	values.Add("columnGroupBy", strings.Join(reportOpts.VariantOption.ColumnGroupBy.List(), ","))
+	for name, variants := range reportOpts.VariantOption.IncludeVariants {
+		for _, v := range variants {
+			values.Add("includeVariant", fmt.Sprintf("%s:%s", name, v))
+		}
+	}
+
+	values.Add("confidence", fmt.Sprintf("%d", reportOpts.AdvancedOption.Confidence))
+	if reportOpts.AdvancedOption.IgnoreDisruption {
+		values.Add("ignoreDisruption", "1")
+	} else {
+		values.Add("ignoreDisruption", "0")
+	}
+	if reportOpts.AdvancedOption.IgnoreMissing {
+		values.Add("ignoreMissing", "1")
+	} else {
+		values.Add("ignoreMissing", "0")
+	}
+	values.Add("minFail", fmt.Sprintf("%d", reportOpts.AdvancedOption.MinimumFailure))
+	values.Add("pity", fmt.Sprintf("%d", reportOpts.AdvancedOption.PityFactor))
+
+	absURL += values.Encode()
+	viewURL := j.sippyURL + "/sippy-ng/component_readiness/main?view=" + view.Name
+	return absURL, viewURL, nil
+}
+
+// createNewJiraIssueForRegressions creates new issue for components by
+// a. setting the ticket's Affects Version/s= sample version.
+// b  adding the Regression label defined by LabelJiraAutomator.
+// c. setting a description with links to CR
+// d. for pre-release, setting "Release Blocker" label to Approved
+func (j JiraAutomator) createNewJiraIssueForRegressions(view crtype.View, component string, tests []crtype.ReportTestSummary, linkedIssue *jira.Issue) (*jira.Issue, error) {
+	if len(tests) > 0 {
+		description := `Component Readiness has found a potential regression in the following tests:`
+		for i, test := range tests {
+			// Only show stats for the worst regression
+			if i == 0 {
+				description += fmt.Sprintf("\n h4. Most Regressed Test:\n{code}%s{code}\n", test.TestName)
+				description += getProbabilityString(test.ReportStatus, *test.FisherExact)
+				description += getStatsString("Sample (being evaluated)", test.SampleStats, view.SampleRelease.RelativeStart, view.SampleRelease.RelativeEnd)
+				description += getStatsString("Base (historical)", *test.BaseStats, view.BaseRelease.RelativeStart, view.BaseRelease.RelativeEnd)
```

Comment:
With the merger of pass rate testing this text gets much more complicated. There may be no basis. There may be no probability string. There *is* however the new "explanations" slice of strings, which the server should populate with WHY it marked this a regression. I did this to hopefully prevent proliferation of if/else logic for how we present this info to users. I suggest scaling back the details here to just show the explanations, avoiding any assumption about fishers.